admin/src/auth/index.ts

```diff
@@ -0,0 +1,103 @@
+import { AuthProvider } from "react-admin";
+
+const localStorageUserKey = "user";
+
+type User = {
+  id: number;
+  username: string;
+  email: string;
+  roles: string[];
+};
+type UserResponse = {
+  error?: string;
+  data?: User | null;
+  token?: string;
+};
+
+async function storeUser(user: User | null) {
+  if (!user) {
+    localStorage.removeItem(localStorageUserKey);
+    throw { status: 401, message: "User not logged in" };
+  } else {
+    localStorage.setItem(localStorageUserKey, JSON.stringify(user));
+  }
+}
+
+export const authProvider: AuthProvider = {
+  // authentication
+  login: async ({ username, password }) => {
+    const response = await fetch("/api/v1/auth/sign-in", {
+      headers: {
+        "Content-Type": "application/json",
+      },
+      method: "POST",
+      body: JSON.stringify({ username, password }),
+    });
+
+    const responseContent: UserResponse = await response.json();
+    console.log(responseContent);
+
+    if (response.status != 200) {
+      throw { message: responseContent.error! };
+    }
+
+    storeUser(responseContent.data!);
+  },
+  checkError: (error) => {
+    const status = error.status;
+    console.log(error);
+    if (status !== 200) {
+      localStorage.removeItem(localStorageUserKey);
+      return Promise.reject();
+    }
+    return Promise.resolve();
+  },
+  checkAuth: async () => {
+    const response = await fetch("/api/v1/auth/check");
```

Comment:
iIn react-admin, checkAuth should only call the server when the token expires. Otherwise, it should be able to determine is the auth is valid by reading the JWT

Reply:
I do not store the JW token client side (it is stored in a secure cookie). But I will use the local storage user for the check).